### DISCUSSION TITLE
2FA OTP issue for Kenya

### DISCUSSION THREAD

**Discussion by sgatibaru** [community/community#89115] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Show & Tell

### Body

I tried using an OTP via my mobile number & noticed the country I'm presently located in, Kenya (+254) isn't available. Could someone/anyone tell me if there's a reason for that or if possible, could it be added to the country list?

**Comment by ghostinhershell** [comment]
Hi @sgatibaru 👋🏾 

Thank so much for writing in. If your country is not on [the list](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/countries-where-sms-authentication-is-supported#supported-countries-for-sms-authentication), then we aren't currently able to reliably deliver text messages to your country. We track delivery rates across all countries, including ones not in the list, and review it to understand when we have to remove countries or can add them back. We do not add countries with low deliverability rates, because it leads to users being locked out of their accounts when they can’t receive the SMS for 2FA.

If GitHub doesn’t support two-factor authentication via text message for your country of residence, you must set up authentication via a TOTP application. For more information on how to configure 2FA, please review our [documentation](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/configuring-two-factor-authentication#configuring-two-factor-authentication-using-a-totp-mobile-app).

You can also find this, and other frequently asked questions, in our [Two Factor Authentication Frequently Asked Questions 🔑 🗝️](https://github.com/orgs/community/discussions/68240) post.

Let me know if you have any additional questions. Thanks!

**Comment by BerjisTech** [comment]
So you can reliably deliver sms to Tanzania and Uganda but not Kenya. Now you're forcing us to enable 2fa without the sms option knowing very well all these so called otp apps will delete your codes whenever they feel like. Won't be the first time, I have been locked out of this account by microsoft auth, I have lost codes by google Auth, lastpass doesn't even work half the time. 

Why is github forcing me to trust my account by a third party, when you can just offer the sms option? Technology is not an issue here and you know it. 

Come August 5th what happens then??????? AUTH APPS ARE NOT RELIABLE, THEY'VE NEVER BEEN

**Comment by lsankona-lyvac** [comment]
I just lost my phone & Laptop and could not regain access to my account since my recovery codes were not synced to my cloud.  As a result of this i have lost quite a lot of work since the inception of my account. This could easily have been averted if SMS was allowed as a 2FA option for kenya. It does not make sense that SMS delivery rates in kenya are unreliable whereas companies like Google,Slack,Meta and numerous banks both local and international support this as an option and there have not been any complaints.
